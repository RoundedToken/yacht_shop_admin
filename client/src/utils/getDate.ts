export function getDate(isoDate: string): string {
    const date = new Date(isoDate);

    return date.toLocaleString('default', {
        day: '2-digit',
        month: 'short',
        year: 'numeric',
        hour: 'numeric',
        minute: 'numeric',
        hour12: false,
    });
}
