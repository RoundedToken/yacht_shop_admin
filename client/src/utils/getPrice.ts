export const getPrice = (price: number) => {
    const formatter = new Intl.NumberFormat('et-EE', {
        style: 'currency',
        currency: 'EUR',
    });
    return formatter.format(price);
};
