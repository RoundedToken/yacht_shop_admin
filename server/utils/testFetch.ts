export class testFetch {
    private endpoint: string;

    constructor(endpoint: string) {
        this.endpoint = endpoint;
    }

    async getData(params: Record<string, string>) {
        const baseUrl = 'http://192.168.1.37:5000/api';
        const url = baseUrl + this.endpoint + '?' + new URLSearchParams(params);

        const response = await fetch(url);

        return await response.json();
    }
}
