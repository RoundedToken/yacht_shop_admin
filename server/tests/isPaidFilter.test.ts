import { describe, expect, test } from 'bun:test';
import { testFetch } from '../utils/testFetch';

describe('Is Paid Filter', () => {
    const getSales = new testFetch('/get_sales');
    const getProducts = new testFetch('/get_products');

    test('Get Sales isPaid: false', async () => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":false,"product":null,"client":null}',
        };
        const data = await getSales.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Products isPaid: false', async () => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":false,"product":null,"client":null}',
        };
        const data = await getProducts.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Sales isPaid: true', async () => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":true,"product":null,"client":null}',
        };
        const data = await getSales.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Products isPaid: true', async () => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":true,"product":null,"client":null}',
        };
        const data = await getProducts.getData(params);

        expect(data).toMatchSnapshot();
    });
});
