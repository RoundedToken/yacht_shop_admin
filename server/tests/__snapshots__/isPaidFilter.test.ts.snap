// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Is Paid Filter Get Sales isPaid: false 1`] = `
{
  "clients": [
    {
      "id": null,
      "name": null,
    },
    {
      "id": 245,
      "name": "Щербаков Антон Васильевич",
    },
    {
      "id": 627,
      "name": "Sailtech Oy (Финляндия)",
    },
    {
      "id": 676,
      "name": "Maritim-Finland",
    },
    {
      "id": 940,
      "name": "web",
    },
    {
      "id": 1108,
      "name": "TREM",
    },
    {
      "id": 1702,
      "name": "MarineTech (Germany)",
    },
    {
      "id": 2766,
      "name": "Optiparts",
    },
    {
      "id": 2773,
      "name": "Windex (Sweden)",
    },
    {
      "id": 9901,
      "name": "Anton Scherbakov",
    },
    {
      "id": 11182,
      "name": "_jamuuk",
    },
    {
      "id": 11183,
      "name": "PJK",
    },
    {
      "id": 11185,
      "name": "Regate (LV)",
    },
    {
      "id": 11186,
      "name": "TommyyLyy",
    },
    {
      "id": 11187,
      "name": "Pro Marine Trade",
    },
    {
      "id": 11188,
      "name": "Татьяна Каверина (Armastus)",
    },
    {
      "id": 11189,
      "name": "NORA LII SOPRADE SELTS",
    },
    {
      "id": 11190,
      "name": "Andres",
    },
    {
      "id": 11191,
      "name": "Nautix",
    },
    {
      "id": 11192,
      "name": "Torivallavalitsus",
    },
    {
      "id": 11193,
      "name": "Ninne",
    },
    {
      "id": 11194,
      "name": "Urmas Varris",
    },
    {
      "id": 11195,
      "name": "WindWalker",
    },
    {
      "id": 11196,
      "name": "Via Grupp OU (Командор)",
    },
    {
      "id": 11201,
      "name": "Tõnis",
    },
    {
      "id": 11202,
      "name": "Jaanus",
    },
    {
      "id": 11203,
      "name": "Meister Arle OU",
    },
    {
      "id": 11204,
      "name": "Prime Marine Service OU",
    },
    {
      "id": 11205,
      "name": "Viking",
    },
    {
      "id": 11206,
      "name": "Saku",
    },
    {
      "id": 11207,
      "name": "SineneLind",
    },
    {
      "id": 11208,
      "name": "Alex Tido",
    },
    {
      "id": 11209,
      "name": "AZZURRA",
    },
    {
      "id": 11210,
      "name": "квас на пиво",
    },
    {
      "id": 11211,
      "name": "MIA (Tallin)",
    },
    {
      "id": 11212,
      "name": "MINNI",
    },
    {
      "id": 11213,
      "name": "Jonпgert OU",
    },
    {
      "id": 11214,
      "name": "MIA Pärnu",
    },
    {
      "id": 11217,
      "name": "Tauno Teluk",
    },
    {
      "id": 11219,
      "name": "Капитан",
    },
    {
      "id": 11220,
      "name": "Johannes  Atlandi Võlg",
    },
    {
      "id": 11221,
      "name": "Nimi",
    },
    {
      "id": 11222,
      "name": "Viahooldus ОÜ",
    },
    {
      "id": 11223,
      "name": "Kaupmees",
    },
    {
      "id": 11224,
      "name": "Reino",
    },
    {
      "id": 11225,
      "name": "Rubit OU",
    },
    {
      "id": 11226,
      "name": "Aleksey  Rusetsky",
    },
    {
      "id": 11227,
      "name": "Corinthian Events OÜ",
    },
    {
      "id": 11228,
      "name": "Ronstan DK",
    },
    {
      "id": 11229,
      "name": "Narva-Joesuu PurjeSpoordiKool",
    },
    {
      "id": 11230,
      "name": "Saaremaa Meri Spordiselts",
    },
    {
      "id": 11231,
      "name": "Tallux Orto",
    },
    {
      "id": 11232,
      "name": "Uku Kuusk",
    },
    {
      "id": 11234,
      "name": "Bainbridge Inc (UK)",
    },
    {
      "id": 11235,
      "name": "BetaMarine (UK)",
    },
    {
      "id": 11236,
      "name": "Change Consulting OÜ",
    },
    {
      "id": 11237,
      "name": "Rait Maru",
    },
    {
      "id": 11239,
      "name": "Žaklar OU",
    },
    {
      "id": 11240,
      "name": "Comar Systems Limited",
    },
    {
      "id": 11241,
      "name": "Kaidor",
    },
    {
      "id": 11242,
      "name": "MADEN",
    },
    {
      "id": 11243,
      "name": "Ropebox Eesti OÜ",
    },
    {
      "id": 11244,
      "name": "Muri",
    },
    {
      "id": 11245,
      "name": "Carrestoration OU",
    },
    {
      "id": 11246,
      "name": "Siren Sailing OU",
    },
    {
      "id": 11247,
      "name": "Episood OU",
    },
    {
      "id": 11248,
      "name": "Akster Grupp OU",
    },
    {
      "id": 11249,
      "name": "Alter",
    },
    {
      "id": 11250,
      "name": "Ronstan AUS",
    },
    {
      "id": 11251,
      "name": "Innon Finance OU",
    },
    {
      "id": 11252,
      "name": "Lettore OÜ",
    },
    {
      "id": 11253,
      "name": "Navideep Services OÜ",
    },
    {
      "id": 11254,
      "name": "Athaman (Polska)",
    },
    {
      "id": 11255,
      "name": "Lõbusõit OÜ",
    },
    {
      "id": 11256,
      "name": "Aleksei Mikhalev",
    },
    {
      "id": 11257,
      "name": "Ropebox OÜ",
    },
    {
      "id": 11258,
      "name": "Kaubakeskus OÜ",
    },
    {
      "id": 11259,
      "name": "Rein Ottoson OÜ",
    },
    {
      "id": 11260,
      "name": "Robert",
    },
    {
      "id": 11261,
      "name": "Merema OÜ",
    },
    {
      "id": 11262,
      "name": "Alexander Rusetsky",
    },
    {
      "id": 11263,
      "name": "Natalja Rusetskaya",
    },
    {
      "id": 11264,
      "name": "Jahtklubi Dago",
    },
    {
      "id": 11265,
      "name": "Elev Luik",
    },
    {
      "id": 11266,
      "name": "_ЗХ2023",
    },
    {
      "id": 11267,
      "name": "Vsevolod pyramidaprint",
    },
    {
      "id": 11268,
      "name": "Emerald",
    },
    {
      "id": 11270,
      "name": "Strand Sailing Management",
    },
    {
      "id": 11271,
      "name": "Tuule",
    },
    {
      "id": 11272,
      "name": "Meelis Kojamess (Баба Яга)",
    },
    {
      "id": 11273,
      "name": "Andrea FE64",
    },
    {
      "id": 11274,
      "name": "Reilut OÜ",
    },
    {
      "id": 11275,
      "name": "Pärnu cruises",
    },
    {
      "id": 11280,
      "name": "Madis Parts",
    },
    {
      "id": 11281,
      "name": "Tera ehitus OU",
    },
    {
      "id": 11282,
      "name": "Korkvender OU",
    },
    {
      "id": 11291,
      "name": "STRAND PROJEKT OÜ",
    },
    {
      "id": 11292,
      "name": "RTR Tehnika",
    },
    {
      "id": 11293,
      "name": "Eprodukt  OÜ",
    },
    {
      "id": 11294,
      "name": "Madis Parts",
    },
    {
      "id": 11295,
      "name": "SeaKayaking",
    },
    {
      "id": 11296,
      "name": "8 Motor Oy",
    },
    {
      "id": 11297,
      "name": "SKELTMER OÜ",
    },
    {
      "id": 11298,
      "name": "Jäälind",
    },
    {
      "id": 11299,
      "name": "MattChem (France)",
    },
    {
      "id": 11300,
      "name": "Car Restoration OU",
    },
    {
      "id": 11301,
      "name": "Valgeranna Seikluspark OÜ",
    },
    {
      "id": 11302,
      "name": "Kaif OÜ",
    },
    {
      "id": 11303,
      "name": "Julechka",
    },
    {
      "id": 11304,
      "name": "Longboat",
    },
    {
      "id": 11305,
      "name": "SAILING MASSIVE  OÜ",
    },
    {
      "id": 11306,
      "name": "AB Technology",
    },
    {
      "id": 11307,
      "name": "Norman Saarso",
    },
    {
      "id": 11308,
      "name": "Tuisu OÜ",
    },
    {
      "id": 11309,
      "name": "Fairaway",
    },
    {
      "id": 11311,
      "name": "Alibi OÜ",
    },
    {
      "id": 11312,
      "name": "Dirham motorpaad",
    },
    {
      "id": 11313,
      "name": "Niined group",
    },
    {
      "id": 11314,
      "name": "Magna Fide",
    },
    {
      "id": 11315,
      "name": "Корячкин",
    },
    {
      "id": 11317,
      "name": "Konstrado",
    },
    {
      "id": 11318,
      "name": "Vektom OÜ",
    },
    {
      "id": 11319,
      "name": "Kalev Jahtklubi",
    },
    {
      "id": 11320,
      "name": "Marta Vintage OÜ",
    },
  ],
  "orders": [
    {
      "allSum": -912.9856,
      "clientName": "Ronstan AUS",
      "date": "2022-07-27T19:18:00.000Z",
      "id": 205261,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 8,
      "productsCount": 124,
      "productsSum": -912.9856,
      "status": "покупка",
    },
    {
      "allSum": -2830.5905,
      "clientName": "Ronstan AUS",
      "date": "2022-07-01T19:13:00.000Z",
      "id": 205260,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 22,
      "productsCount": 330,
      "productsSum": -2830.5905,
      "status": "покупка",
    },
    {
      "allSum": -397.2849,
      "clientName": "MarineTech (Germany)",
      "date": "2022-09-01T10:09:00.000Z",
      "id": 205215,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 18,
      "productsCount": 822,
      "productsSum": -397.2849,
      "status": "покупка",
    },
    {
      "allSum": 1,
      "clientName": "Urmas Varris",
      "date": "2022-08-28T15:55:00.000Z",
      "id": 205212,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 1,
      "productsCount": -5,
      "productsSum": 1,
      "status": "продажа",
    },
    {
      "allSum": 293.04,
      "clientName": "Jaanus",
      "date": "2022-09-02T15:37:00.000Z",
      "id": 205177,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 14,
      "productsCount": -54,
      "productsSum": 293.04,
      "status": "продажа",
    },
    {
      "allSum": -770.56,
      "clientName": "Alter",
      "date": "2022-08-25T12:05:00.000Z",
      "id": 205176,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 24,
      "productsCount": 47,
      "productsSum": -770.56,
      "status": "покупка",
    },
    {
      "allSum": -2799.5089,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-08-10T15:38:00.000Z",
      "id": 205162,
      "isPaid": false,
      "paidSum": -2799.51,
      "payType": "Б/н",
      "positionCount": 30,
      "productsCount": 4074,
      "productsSum": -2799.5089,
      "status": "покупка",
    },
    {
      "allSum": -1155.7666,
      "clientName": "Ronstan DK",
      "date": "2022-05-06T14:44:00.000Z",
      "id": 205160,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 7,
      "productsCount": 44,
      "productsSum": -1155.7666,
      "status": "покупка",
    },
    {
      "allSum": -303.4701,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-05-26T14:41:00.000Z",
      "id": 205159,
      "isPaid": false,
      "paidSum": -303.47,
      "payType": "Б/н",
      "positionCount": 6,
      "productsCount": 12,
      "productsSum": -303.4701,
      "status": "покупка",
    },
    {
      "allSum": -4676.102,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-05-06T14:15:00.000Z",
      "id": 205158,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 77,
      "productsCount": 236,
      "productsSum": -4676.102,
      "status": "покупка",
    },
  ],
  "pagesCount": 8,
  "rowsCount": 79,
  "totalAllSum": -28471.3804,
  "totalPaidSum": -5667.878,
  "totalProductSum": -28471.3804,
}
`;

exports[`Is Paid Filter Get Products isPaid: false 1`] = `
{
  "pagesCount": 63,
  "products": [
    {
      "avgPrice": 45.83,
      "id": 58222,
      "maxPrice": 45.83,
      "minPrice": 45.83,
      "name": "Откреночные шорты Race Gravity Hiking shorts  JL",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -45.83,
    },
    {
      "avgPrice": 45.83,
      "id": 58221,
      "maxPrice": 45.83,
      "minPrice": 45.83,
      "name": "Откреночные шорты Race Gravity Hiking shorts  JXL",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -45.83,
    },
    {
      "avgPrice": 45.83,
      "id": 58220,
      "maxPrice": 45.83,
      "minPrice": 45.83,
      "name": "Откреночные шорты Race Gravity Hiking shorts  S",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -45.83,
    },
    {
      "avgPrice": 25.21,
      "id": 58219,
      "maxPrice": 25.21,
      "minPrice": 25.21,
      "name": "Очки Longrock Junior  Ash/Smoke",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -25.21,
    },
    {
      "avgPrice": 25.21,
      "id": 58218,
      "maxPrice": 25.21,
      "minPrice": 25.21,
      "name": "Очки Longrock Junior  Black/Smoke",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -25.21,
    },
    {
      "avgPrice": 16.04,
      "id": 58216,
      "maxPrice": 16.04,
      "minPrice": 16.04,
      "name": "Waterproof Beanie Graohite",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -16.04,
    },
    {
      "avgPrice": 22.92,
      "id": 58215,
      "maxPrice": 22.92,
      "minPrice": 22.92,
      "name": "Pro Gloves (Long Finger) - Must - L",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -45.84,
    },
    {
      "avgPrice": 22.92,
      "id": 58214,
      "maxPrice": 22.92,
      "minPrice": 22.92,
      "name": "Pro Gloves (Long Finger) - Must - XL",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -45.84,
    },
    {
      "avgPrice": 18.33,
      "id": 58213,
      "maxPrice": 18.33,
      "minPrice": 18.33,
      "name": "Детская футболка с длинными рукавами Grey Melange,  JL",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -18.33,
    },
    {
      "avgPrice": 41.25,
      "id": 58212,
      "maxPrice": 41.25,
      "minPrice": 41.25,
      "name": "Футболка неопреновая Thermoskin Black/Red XXL",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -41.25,
    },
  ],
  "rowsCount": 621,
}
`;

exports[`Is Paid Filter Get Sales isPaid: true 1`] = `
{
  "clients": [
    {
      "id": null,
      "name": null,
    },
    {
      "id": 245,
      "name": "Щербаков Антон Васильевич",
    },
    {
      "id": 627,
      "name": "Sailtech Oy (Финляндия)",
    },
    {
      "id": 676,
      "name": "Maritim-Finland",
    },
    {
      "id": 940,
      "name": "web",
    },
    {
      "id": 1108,
      "name": "TREM",
    },
    {
      "id": 1702,
      "name": "MarineTech (Germany)",
    },
    {
      "id": 2766,
      "name": "Optiparts",
    },
    {
      "id": 2773,
      "name": "Windex (Sweden)",
    },
    {
      "id": 9901,
      "name": "Anton Scherbakov",
    },
    {
      "id": 11182,
      "name": "_jamuuk",
    },
    {
      "id": 11183,
      "name": "PJK",
    },
    {
      "id": 11185,
      "name": "Regate (LV)",
    },
    {
      "id": 11186,
      "name": "TommyyLyy",
    },
    {
      "id": 11187,
      "name": "Pro Marine Trade",
    },
    {
      "id": 11188,
      "name": "Татьяна Каверина (Armastus)",
    },
    {
      "id": 11189,
      "name": "NORA LII SOPRADE SELTS",
    },
    {
      "id": 11190,
      "name": "Andres",
    },
    {
      "id": 11191,
      "name": "Nautix",
    },
    {
      "id": 11192,
      "name": "Torivallavalitsus",
    },
    {
      "id": 11193,
      "name": "Ninne",
    },
    {
      "id": 11194,
      "name": "Urmas Varris",
    },
    {
      "id": 11195,
      "name": "WindWalker",
    },
    {
      "id": 11196,
      "name": "Via Grupp OU (Командор)",
    },
    {
      "id": 11201,
      "name": "Tõnis",
    },
    {
      "id": 11202,
      "name": "Jaanus",
    },
    {
      "id": 11203,
      "name": "Meister Arle OU",
    },
    {
      "id": 11204,
      "name": "Prime Marine Service OU",
    },
    {
      "id": 11205,
      "name": "Viking",
    },
    {
      "id": 11206,
      "name": "Saku",
    },
    {
      "id": 11207,
      "name": "SineneLind",
    },
    {
      "id": 11208,
      "name": "Alex Tido",
    },
    {
      "id": 11209,
      "name": "AZZURRA",
    },
    {
      "id": 11210,
      "name": "квас на пиво",
    },
    {
      "id": 11211,
      "name": "MIA (Tallin)",
    },
    {
      "id": 11212,
      "name": "MINNI",
    },
    {
      "id": 11213,
      "name": "Jonпgert OU",
    },
    {
      "id": 11214,
      "name": "MIA Pärnu",
    },
    {
      "id": 11217,
      "name": "Tauno Teluk",
    },
    {
      "id": 11219,
      "name": "Капитан",
    },
    {
      "id": 11220,
      "name": "Johannes  Atlandi Võlg",
    },
    {
      "id": 11221,
      "name": "Nimi",
    },
    {
      "id": 11222,
      "name": "Viahooldus ОÜ",
    },
    {
      "id": 11223,
      "name": "Kaupmees",
    },
    {
      "id": 11224,
      "name": "Reino",
    },
    {
      "id": 11225,
      "name": "Rubit OU",
    },
    {
      "id": 11226,
      "name": "Aleksey  Rusetsky",
    },
    {
      "id": 11227,
      "name": "Corinthian Events OÜ",
    },
    {
      "id": 11228,
      "name": "Ronstan DK",
    },
    {
      "id": 11229,
      "name": "Narva-Joesuu PurjeSpoordiKool",
    },
    {
      "id": 11230,
      "name": "Saaremaa Meri Spordiselts",
    },
    {
      "id": 11231,
      "name": "Tallux Orto",
    },
    {
      "id": 11232,
      "name": "Uku Kuusk",
    },
    {
      "id": 11234,
      "name": "Bainbridge Inc (UK)",
    },
    {
      "id": 11235,
      "name": "BetaMarine (UK)",
    },
    {
      "id": 11236,
      "name": "Change Consulting OÜ",
    },
    {
      "id": 11237,
      "name": "Rait Maru",
    },
    {
      "id": 11239,
      "name": "Žaklar OU",
    },
    {
      "id": 11240,
      "name": "Comar Systems Limited",
    },
    {
      "id": 11241,
      "name": "Kaidor",
    },
    {
      "id": 11242,
      "name": "MADEN",
    },
    {
      "id": 11243,
      "name": "Ropebox Eesti OÜ",
    },
    {
      "id": 11244,
      "name": "Muri",
    },
    {
      "id": 11245,
      "name": "Carrestoration OU",
    },
    {
      "id": 11246,
      "name": "Siren Sailing OU",
    },
    {
      "id": 11247,
      "name": "Episood OU",
    },
    {
      "id": 11248,
      "name": "Akster Grupp OU",
    },
    {
      "id": 11249,
      "name": "Alter",
    },
    {
      "id": 11250,
      "name": "Ronstan AUS",
    },
    {
      "id": 11251,
      "name": "Innon Finance OU",
    },
    {
      "id": 11252,
      "name": "Lettore OÜ",
    },
    {
      "id": 11253,
      "name": "Navideep Services OÜ",
    },
    {
      "id": 11254,
      "name": "Athaman (Polska)",
    },
    {
      "id": 11255,
      "name": "Lõbusõit OÜ",
    },
    {
      "id": 11256,
      "name": "Aleksei Mikhalev",
    },
    {
      "id": 11257,
      "name": "Ropebox OÜ",
    },
    {
      "id": 11258,
      "name": "Kaubakeskus OÜ",
    },
    {
      "id": 11259,
      "name": "Rein Ottoson OÜ",
    },
    {
      "id": 11260,
      "name": "Robert",
    },
    {
      "id": 11261,
      "name": "Merema OÜ",
    },
    {
      "id": 11262,
      "name": "Alexander Rusetsky",
    },
    {
      "id": 11263,
      "name": "Natalja Rusetskaya",
    },
    {
      "id": 11264,
      "name": "Jahtklubi Dago",
    },
    {
      "id": 11265,
      "name": "Elev Luik",
    },
    {
      "id": 11266,
      "name": "_ЗХ2023",
    },
    {
      "id": 11267,
      "name": "Vsevolod pyramidaprint",
    },
    {
      "id": 11268,
      "name": "Emerald",
    },
    {
      "id": 11270,
      "name": "Strand Sailing Management",
    },
    {
      "id": 11271,
      "name": "Tuule",
    },
    {
      "id": 11272,
      "name": "Meelis Kojamess (Баба Яга)",
    },
    {
      "id": 11273,
      "name": "Andrea FE64",
    },
    {
      "id": 11274,
      "name": "Reilut OÜ",
    },
    {
      "id": 11275,
      "name": "Pärnu cruises",
    },
    {
      "id": 11280,
      "name": "Madis Parts",
    },
    {
      "id": 11281,
      "name": "Tera ehitus OU",
    },
    {
      "id": 11282,
      "name": "Korkvender OU",
    },
    {
      "id": 11291,
      "name": "STRAND PROJEKT OÜ",
    },
    {
      "id": 11292,
      "name": "RTR Tehnika",
    },
    {
      "id": 11293,
      "name": "Eprodukt  OÜ",
    },
    {
      "id": 11294,
      "name": "Madis Parts",
    },
    {
      "id": 11295,
      "name": "SeaKayaking",
    },
    {
      "id": 11296,
      "name": "8 Motor Oy",
    },
    {
      "id": 11297,
      "name": "SKELTMER OÜ",
    },
    {
      "id": 11298,
      "name": "Jäälind",
    },
    {
      "id": 11299,
      "name": "MattChem (France)",
    },
    {
      "id": 11300,
      "name": "Car Restoration OU",
    },
    {
      "id": 11301,
      "name": "Valgeranna Seikluspark OÜ",
    },
    {
      "id": 11302,
      "name": "Kaif OÜ",
    },
    {
      "id": 11303,
      "name": "Julechka",
    },
    {
      "id": 11304,
      "name": "Longboat",
    },
    {
      "id": 11305,
      "name": "SAILING MASSIVE  OÜ",
    },
    {
      "id": 11306,
      "name": "AB Technology",
    },
    {
      "id": 11307,
      "name": "Norman Saarso",
    },
    {
      "id": 11308,
      "name": "Tuisu OÜ",
    },
    {
      "id": 11309,
      "name": "Fairaway",
    },
    {
      "id": 11311,
      "name": "Alibi OÜ",
    },
    {
      "id": 11312,
      "name": "Dirham motorpaad",
    },
    {
      "id": 11313,
      "name": "Niined group",
    },
    {
      "id": 11314,
      "name": "Magna Fide",
    },
    {
      "id": 11315,
      "name": "Корячкин",
    },
    {
      "id": 11317,
      "name": "Konstrado",
    },
    {
      "id": 11318,
      "name": "Vektom OÜ",
    },
    {
      "id": 11319,
      "name": "Kalev Jahtklubi",
    },
    {
      "id": 11320,
      "name": "Marta Vintage OÜ",
    },
  ],
  "orders": [
    {
      "allSum": 624.2103,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-09-25T12:02:00.000Z",
      "id": 205282,
      "isPaid": true,
      "paidSum": 624.2103,
      "payType": "Visa, Нал",
      "positionCount": 3,
      "productsCount": -15,
      "productsSum": 624.2103,
      "status": "продажа",
    },
    {
      "allSum": 65,
      "clientName": "Andres",
      "date": "2022-10-02T16:40:00.000Z",
      "id": 205277,
      "isPaid": true,
      "paidSum": 65,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 65,
      "status": "продажа",
    },
    {
      "allSum": 4.1,
      "clientName": "_jamuuk",
      "date": "2022-09-17T12:21:00.000Z",
      "id": 205270,
      "isPaid": true,
      "paidSum": 4.1,
      "payType": "Нал",
      "positionCount": 2,
      "productsCount": -2,
      "productsSum": 4.1,
      "status": "продажа",
    },
    {
      "allSum": 5.4,
      "clientName": "Elev Luik",
      "date": "2022-09-17T11:56:00.000Z",
      "id": 205269,
      "isPaid": true,
      "paidSum": 5.4,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 5.4,
      "status": "продажа",
    },
    {
      "allSum": 31,
      "clientName": "_jamuuk",
      "date": "2022-09-17T10:56:00.000Z",
      "id": 205267,
      "isPaid": true,
      "paidSum": 31,
      "payType": "Visa",
      "positionCount": 2,
      "productsCount": -2,
      "productsSum": 31,
      "status": "продажа",
    },
    {
      "allSum": 60,
      "clientName": "_jamuuk",
      "date": "2022-09-11T16:55:00.000Z",
      "id": 205266,
      "isPaid": true,
      "paidSum": 60,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 60,
      "status": "продажа",
    },
    {
      "allSum": 15.5,
      "clientName": "_jamuuk",
      "date": "2022-09-11T16:39:00.000Z",
      "id": 205265,
      "isPaid": true,
      "paidSum": 15.5,
      "payType": "Visa",
      "positionCount": 3,
      "productsCount": -3,
      "productsSum": 15.5,
      "status": "продажа",
    },
    {
      "allSum": 12,
      "clientName": "_jamuuk",
      "date": "2022-09-11T16:22:00.000Z",
      "id": 205264,
      "isPaid": true,
      "paidSum": 12,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -1,
      "productsSum": 12,
      "status": "продажа",
    },
    {
      "allSum": 25,
      "clientName": "_jamuuk",
      "date": "2022-09-11T16:08:00.000Z",
      "id": 205263,
      "isPaid": true,
      "paidSum": 25,
      "payType": "Visa, Нал",
      "positionCount": 2,
      "productsCount": -2,
      "productsSum": 25,
      "status": "продажа",
    },
    {
      "allSum": 477.5,
      "clientName": "_jamuuk",
      "date": "2022-09-10T17:56:00.000Z",
      "id": 205256,
      "isPaid": true,
      "paidSum": 477.5,
      "payType": "Visa",
      "positionCount": 11,
      "productsCount": -33,
      "productsSum": 477.5,
      "status": "продажа",
    },
  ],
  "pagesCount": 99,
  "rowsCount": 989,
  "totalAllSum": 34815.9318,
  "totalPaidSum": 36054.0886,
  "totalProductSum": 34815.9318,
}
`;

exports[`Is Paid Filter Get Products isPaid: true 1`] = `
{
  "pagesCount": 110,
  "products": [
    {
      "avgPrice": 200.6273,
      "id": 58224,
      "maxPrice": 231.0589,
      "minPrice": 170.1958,
      "name": "DRIVEPLATE SAE6.5 -HURTH / PRM 22-B-60",
      "ordersCount": 2,
      "totalCount": 0,
      "totalSum": 60.8631,
    },
    {
      "avgPrice": 15.3524,
      "id": 58223,
      "maxPrice": 17.6811,
      "minPrice": 13.0237,
      "name": "Pa PRESSURE CAP TWIN SEAL",
      "ordersCount": 2,
      "totalCount": 0,
      "totalSum": 9.3148,
    },
    {
      "avgPrice": 10,
      "id": 58210,
      "maxPrice": 10,
      "minPrice": 10,
      "name": "Огон дайнима 3 мм",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 10,
    },
    {
      "avgPrice": 16.1326,
      "id": 58208,
      "maxPrice": 20,
      "minPrice": 12.2652,
      "name": "wireless battery charger water resistant",
      "ordersCount": 2,
      "totalCount": 1,
      "totalSum": -4.5304,
    },
    {
      "avgPrice": 67.358,
      "id": 58207,
      "maxPrice": 67.358,
      "minPrice": 67.358,
      "name": "Retractable telescopic ladders with handles",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -67.358,
    },
    {
      "avgPrice": 6.1326,
      "id": 58206,
      "maxPrice": 6.1326,
      "minPrice": 6.1326,
      "name": "Коннектор берегового питания CE/SCHUKO",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -12.2652,
    },
    {
      "avgPrice": 8.0427,
      "id": 58205,
      "maxPrice": 8.0427,
      "minPrice": 8.0427,
      "name": "CE/SCHUKO Extension cable",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -16.0854,
    },
    {
      "avgPrice": 8.5231,
      "id": 58204,
      "maxPrice": 8.5231,
      "minPrice": 8.5231,
      "name": "LED anchor light with base «EASY LED» white colour",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -17.0462,
    },
    {
      "avgPrice": 74.4768,
      "id": 58201,
      "maxPrice": 74.4768,
      "minPrice": 74.4768,
      "name": "Закрутка геннакера серия 60. верхний шарнир",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 148.9536,
    },
    {
      "avgPrice": 121.3006,
      "id": 58200,
      "maxPrice": 121.3006,
      "minPrice": 121.3006,
      "name": "Закрутка для геннакера серия 60. нижняя часть",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 242.6012,
    },
  ],
  "rowsCount": 1092,
}
`;
